x-pack/plugins/security_solution/public/common/containers/first_last_seen/index.tsx

```diff
@@ -10,78 +10,77 @@ import { useCallback, useEffect, useRef, useState } from 'react';
 import { Subscription } from 'rxjs';
 
 import { isCompleteResponse, isErrorResponse } from '@kbn/data-plugin/common';
-import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
-import { useKibana } from '../../../../common/lib/kibana';
-import {
-  HostsQueries,
-  HostFirstLastSeenStrategyResponse,
-  HostFirstLastSeenRequestOptions,
-} from '../../../../../common/search_strategy/security_solution';
+import { useAppToasts } from '../../hooks/use_app_toasts';
+import { useKibana } from '../../lib/kibana';
 
 import * as i18n from './translations';
-import { Direction, DocValueFields } from '../../../../../common/search_strategy';
+import {
+  Direction,
+  FirstLastSeenQuery,
+  FirstLastSeenRequestOptions,
+  FirstLastSeenStrategyResponse,
+} from '../../../../common/search_strategy';
+import { useSourcererDataView } from '../sourcerer';
 
-const ID = 'firstLastSeenHostQuery';
+const ID = 'firstLastSeenQuery';
 
-export interface FirstLastSeenHostArgs {
+export interface FirstLastSeenArgs {
   id: string;
   errorMessage: string | null;
   firstSeen?: string | null;
   lastSeen?: string | null;
   order: Direction.asc | Direction.desc | null;
 }
-interface UseHostFirstLastSeen {
-  docValueFields: DocValueFields[];
-  hostName: string;
-  indexNames: string[];
+interface UseFirstLastSeen {
+  field: string;
+  value: string;
   order: Direction.asc | Direction.desc;
 }
 
-export const useFirstLastSeenHost = ({
-  docValueFields,
-  hostName,
-  indexNames,
+export const useFirstLastSeen = ({
+  field,
+  value,
   order,
-}: UseHostFirstLastSeen): [boolean, FirstLastSeenHostArgs] => {
-  const { data } = useKibana().services;
+}: UseFirstLastSeen): [boolean, FirstLastSeenArgs] => {
+  const { search } = useKibana().services.data;
   const abortCtrl = useRef(new AbortController());
   const searchSubscription$ = useRef(new Subscription());
   const [loading, setLoading] = useState(false);
-  const [firstLastSeenHostRequest, setFirstLastSeenHostRequest] =
-    useState<HostFirstLastSeenRequestOptions>({
-      defaultIndex: indexNames,
-      docValueFields: docValueFields ?? [],
-      factoryQueryType: HostsQueries.firstOrLastSeen,
-      hostName,
-      order,
-    });
+  const { docValueFields, selectedPatterns: defaultIndex } = useSourcererDataView();
```

Comment:
I moved this into the hook, before it was being called outside of the component and being passed through and I didn't see any reason for that. Of course let me know if I am missing something.

Reply:
ok for now but will have to be part of refactoring for `useSearchStrategy`:  https://github.com/elastic/kibana/issues/129054

Reply:
see here: https://github.com/elastic/kibana/pull/133539/files#r895938972

Reply:
fixed!